test/core/rename/ROBSinkUnit.hpp

```diff
@@ -0,0 +1,96 @@
+
+#pragma once
+
+#include "core/InstGenerator.hpp"
+
+#include "sparta/simulation/TreeNode.hpp"
+#include "sparta/statistics/StatisticSet.hpp"
+#include "sparta/events/SingleCycleUniqueEvent.hpp"
+#include "sparta/utils/SpartaSharedPointer.hpp"
+#include "sparta/utils/LogUtils.hpp"
+
+#include <string>
+
+namespace rename_test
+{
+
+    ////////////////////////////////////////////////////////////////////////////////
+    // "Sink" unit, just sinks instructions sent to it.  Sends credits
+    // back as directed by params/execution mode
+    class ROBSinkUnit : public sparta::Unit
```

Comment:
If this class is mostly the same as `test/core/common/SinkUnit.hpp` consider deriving from it instead of copying the entire class.

Reply:
Addressed in [163a70b](https://github.com/riscv-software-src/riscv-perf-model/pull/40/commits/163a70b5862658deca0df16b035bbe59dc841c92). I had to create a separate port/function to mark instructions being sunk to retired, but that allows ROBSinkUnit to fully inherit from SinkUnit.